assets/js/data/schema/selectors.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * External dependencies
+ */
+import { sprintf } from '@wordpress/i18n';
+import { createRegistrySelector } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY } from './constants';
+
+/**
+ * Returns the requested route for the given arguments.
+ *
+ * @param {Object} state     The original state.
+ * @param {string} namespace The namespace for the route.
+ * @param {string} modelName The model being requested (i.e. products/attributes)
+ * @param {Array}  [ids]     This is for any ids that might be implemented in
+ *                           the route request. It is not for any query
+ *                           parameters.
+ *
+ * Ids example:
+ * If you are looking for the route for a single product on the `wc/blocks`
+ * namespace, then you'd have `[ 20 ]` as the ids.  This would produce something
+ * like `/wc/blocks/products/20`
+ *
+ *
+ * @throws {Error}  If there is no route for the given arguments, then this will
+ *                  throw
+ *
+ * @return {string} The route if it is available.
+ */
+export const getRoute = createRegistrySelector(
+	( select ) => ( state, namespace, modelName, ids = [] ) => {
+		const hasResolved = select( STORE_KEY ).hasFinishedResolution(
+			'getRoutes',
+			[ namespace ]
+		);
```

Comment:
This will allow for only throwing an error if the `getRoutes` selector has been resolved for the given namespace.  It's a way of ensuring that if developers have a typo in any of the arguments, the problem surfaces immediately.  Client code can optionally catch the thrown errors in cases where a route may be conditionally present depending on plugin/extension being active or not but that's not the responsibility of this code block.